cmd/dockerd/daemon_windows.go

```diff
@@ -89,9 +89,18 @@ func newCgroupParent(config *config.Config) string {
 	return ""
 }
 
-func (cli *DaemonCli) initContainerd(_ context.Context) (func(time.Duration) error, error) {
-	system.InitContainerdRuntime(cli.ContainerdAddr)
-	return nil, nil
+func (cli *DaemonCli) initContainerd(ctx context.Context) (func(time.Duration) error, error) {
+	if cli.ContainerdAddr != "" {
+		// use system containerd at the given address.
+		system.InitContainerdRuntime(cli.ContainerdAddr)
+		return nil, nil
```

Comment:
Currently all this does is set a flag if the supplied path is non-empty, i.e. it's always the case here. It should probably just unconditionally set the flag, be named better, and also called if `cli.initializeContainerd(ctx)` below succeeds (which should fix the issue noted in https://github.com/moby/moby/pull/47955#issuecomment-2180056463).

It's only used to implement `system.ContainerdRuntimeSupported()` work, so maybe there's a better way to signal that after these changes?

Maybe just starting this function with something like
```golang
defer func() { system.InitContainerdRuntime(cli.ContainerdAddr) }()
```

would work? That way, however we leave this function, the final value of `cli.ContainerdAddr` is used to determine the value of `system.ContainerdRuntimeSupported()`.